src/client/js/client.js

```diff
@@ -1 +1,495 @@
+			link = link.split('?');
+			var _name = link[1] || "";
+			link = link[0];
+			link += "?'" + name + "':" + num + "," + ind;
+
+			txt = txt.replace(a, name);
+			var ment = this.getMentionsItem(_name, '', link);
+			arr.push(ment);
+		}
+		return txt;
+	}
+
+	addCoreBlock(json, txt, num) {
+		if (!txt) return num;
+		var blocks = txt.split(this.state.endHeader);
+
+		var name = blocks.length == 2 ? blocks[0] : '';
+		name = this.checkMention(json.mentions, name, 1);
+		name = this.replaceLineFeed(name);
+
+		json.name = name;
+
+		var ps = blocks[blocks.length - 1].split('<br>');
+		ps = this.normalizeOUL(ps);
+		for (var i = 0; i < ps.length; i++) {
+			if (ps[i].trim() !== "") {
+				ps[i] = ps[i].replace(/&nbsp;/gi, ' ');
+				var p = this.checkMention(json.mentions, ps[i], num + 2);
+				p = this.replaceLineFeed(p);
+				json.articleBody.push(p);
+			}
+			num += 1;
+		}
+
+		return num;
+	}
+
+	addParagraph(json, txt, num) {
+		if (!txt) return num;
+		var regHttp = /(https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*))/i;
+		var regTitle = /<a [^>]+>([^<]+)<\/a>/i;
+		var regHref = /href=["|']([^'"]+)/i;
+		var blocks = txt.split(this.state.endHeader);
+		
+		var name = blocks[0];
+		name = this.checkMention(json.mentions, name, num + 1);
+		name = this.replaceLineFeed(name);
+		
+		var part = this.getPart(name);
+
+		var ps = blocks[1].split('<br>');
+		ps = this.normalizeOUL(ps);
+		for (var i = 0; i < ps.length; i++) {
+			if (ps[i].trim() !== "") {
+			 	if (ps[i].replace(regHttp, "").trim() !== "" ) {
+					part.articleBody = part.articleBody || [];
+					ps[i] = ps[i].replace(/&nbsp;/gi, ' ');
+					var p = this.checkMention(json.mentions, ps[i], num + 2);
+					p = this.replaceLineFeed(p);
+					part.articleBody.push(p);
+					num += 1;
+				} else {
+					part.url = regHref.exec(ps[i]) ? regHref.exec(ps[i])[1] : regHttp.exec(ps[i])[1];
+				}
+			} else {
+				num += 1;
+			}
+		}
+
+		json.hasPart.push(part);
+
+		return num;
+	}
+
+	save() {
+		var widgetData = $(this.state.$textarea_widget)
+			.val();
+		var txt = $(this.state.$textarea)
+			.val();
+		var json = this.calculateJson(txt, widgetData);
+		if (!json) return;
+
+		var textToWrite = this.state.cleshe.replace('|BODY|', JSON.stringify(json));
+		textToWrite = textToWrite.replace('|TITLE|', json.name);
+
+
+
+		//ToDo: test
+		//console.log(textToWrite);
+		var url = this.state.saveUrl;
+		var contents = "<html></html>";
+		$.ajax({
+				url: "http://storage."+domain+"/api/save",
+				type: 'post',
+				'dataType': 'json',
+				data: {
+					'url': url,
+					'bodyData': textToWrite
+				},
+				xhrFields: {
+					withCredentials: true
+				}
+			})
+			.success((res) => {
+				window.location = res.url;
+			});
+	};
+
+	disableSave() {
+		$('#save-button-id div a').addClass('disabled');
+		$('#save-button-id div').attr('title', 'You can\'t save to non webrunes hosted page. Use \"Save As\" and upload file manually');
```

Comment:
You can only save the page at webrunes.com. Use "Save as" and upload file manually.